src/python/gudhi/simplex_tree.pyx

```diff
@@ -396,6 +396,66 @@ cdef class SimplexTree:
         """
         return self.get_ptr().make_filtration_non_decreasing()
 
+    def extend_filtration(self):
+        """ Extend filtration for computing extended persistence. This function only uses the 
+        filtration values at the 0-dimensional simplices, and computes the extended persistence 
+        diagram induced by the lower-star filtration computed with these values. 
+
+        .. note::
+
+            Note that after calling this function, the filtration 
+            values are actually modified within the Simplex_tree. 
+            The function :func:`extended_persistence()<gudhi.SimplexTree.extended_persistence>`
+            retrieves the original values.
+
+        .. note::
+
+            Note that this code creates an extra vertex internally, so you should make sure that
+            the Simplex_tree does not contain a vertex with the largest Vertex_handle. 
+        """
+        return self.get_ptr().compute_extended_filtration()
+
+    def extended_persistence(self, homology_coeff_field=11, min_persistence=0, persistence_dim_max = False):
+        """This function retrieves good values for extended persistence, and separate the diagrams 
+        into the ordinary, relative, extended+ and extended- subdiagrams.
+
+        :param homology_coeff_field: The homology coefficient field. Must be a
+            prime number. Default value is 11.
+        :type homology_coeff_field: int.
+        :param min_persistence: The minimum persistence value to take into
+            account (strictly greater than min_persistence). Default value is
+            0.0.
+            Sets min_persistence to -1.0 to see all values.
+        :type min_persistence: float.
+        :param persistence_dim_max: If true, the persistent homology for the
```

Comment:
The default of False is mostly because of Rips, alpha-complex, etc, where homology in the maximal dimension is seldom interesting. Here, we know that we are dealing with a lower-star filtration. Do you expect that we will usually want homology in the maximal dimension or not? If the complex is a graph, after coning, the maximal dimension is 2, but I don't see how we could ever have any H_2 in the cone. So maybe we shouldn't provide the option at all?

Reply:
Yeah now that you mention it, it makes a lot of sense. Fixed it.

Reply:
Er, shouldn't we pass `False` if there is never any homology in the maximum dimension, to save computation time?

Reply:
Oops, yes you are right. Done.